### PR TITLE
Add an IP version of the transport

### PR DESCRIPTION
**Pull request by AnthonySteele**
Useful for lambdas

Want review before merging, on suitability for using in C# AWS lambdas.

### CODE REVIEW

File: src/JustEat.StatsD/IpTransport.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        private readonly IPEndPointSource _endpointSource;
+
+        public IpTransport(IPEndPointSource endPointSource)
+        {
+            if (endPointSource == null)
+            {
+                throw new ArgumentNullException(nameof(endPointSource));
+            }
+            _endpointSource = endPointSource;
```
Comment (martincostello) [inline src/JustEat.StatsD/IpTransport.cs]: Now we're on VS 2017 for this, could write this as:

```csharp
_endpointSource = endPointSource ?? throw new ArgumentNullException(nameof(endPointSource));
```

Comment (AnthonySteele) [inline src/JustEat.StatsD/IpTransport.cs]: We _can_ do that in a couple of places. it seemed a little cutting-edge for me at the time, but if you like I can change it.

Comment (martincostello) [inline src/JustEat.StatsD/IpTransport.cs]: Just a suggestion as this is new. Not essential, but would make it a bit more terse.

Comment (AnthonySteele) [inline src/JustEat.StatsD/IpTransport.cs]: Maybe in  a seperate PR, as I would apply it across all classes.

Comment (AnthonySteele) [inline src/JustEat.StatsD/IpTransport.cs]: I suggest that a review of using new syntax throughout be left to a later PR. But let's actually do it some time soon.

Comment (martincostello) [inline src/JustEat.StatsD/IpTransport.cs]: 👍

**Pull request merged**